packages/wallets/coin98/src/adapter.ts

```diff
@@ -140,4 +147,13 @@ export class Coin98WalletAdapter extends BaseSignerWalletAdapter {
         }
         return signedTransactions;
     }
+
+    private _accountChanged = () => {
+        const wallet = this._wallet;
+        console.log(wallet);
+        if (wallet) {
+            wallet.off('accountsChanged', this._accountChanged);
```

Comment:
Why does this remove the event listener? If it's expecting a listener to disconnect and reconnect, that's prescribing too much behavior, and it's going to produce an inconsistent state in the adapter (no listener) if that listener doesn't disconnect.